tracer/src/Datadog.Trace/Propagators/W3CTraceContextPropagator.cs

```diff
@@ -7,125 +7,434 @@
 
 using System;
 using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Text;
+using Datadog.Trace.Util;
 
 namespace Datadog.Trace.Propagators
 {
+    // https://www.w3.org/TR/trace-context/
     internal class W3CTraceContextPropagator : IContextInjector, IContextExtractor
     {
         /// <summary>
-        /// W3C TraceParent header
+        /// W3C traceparent header name
         /// </summary>
-        public const string TraceParent = "traceparent";
+        public const string TraceParentHeaderName = "traceparent";
+
+        /// <summary>
+        /// W3C tracestate header name
+        /// </summary>
+        public const string TraceStateHeaderName = "tracestate";
 
         public static readonly W3CTraceContextPropagator Instance = new();
 
         public void Inject<TCarrier, TCarrierSetter>(SpanContext context, TCarrier carrier, TCarrierSetter carrierSetter)
             where TCarrierSetter : struct, ICarrierSetter<TCarrier>
         {
-            var traceId = IsValidTraceId(context.RawTraceId) ? context.RawTraceId : context.TraceId.ToString("x32");
-            var spanId = IsValidSpanId(context.RawSpanId) ? context.RawSpanId : context.SpanId.ToString("x16");
-            var samplingPriority = context.TraceContext?.SamplingPriority ?? context.SamplingPriority;
+            var traceparent = CreateTraceParentHeader(context);
+
+            if (!string.IsNullOrWhiteSpace(traceparent))
+            {
+                carrierSetter.Set(carrier, TraceParentHeaderName, traceparent);
+            }
```

Comment:
nit: AFAICT, traceparent is never null or whitespace here
```suggestion
            var traceparent = CreateTraceParentHeader(context);
            carrierSetter.Set(carrier, TraceParentHeaderName, traceparent);
```

Reply:
Correct, thanks.